### PR TITLE
EP/MIN-278 Liquidate insolvent loan

### PR DESCRIPTION
**Pull request by Apollo50**
**Description:**

Fix logic of liquidation.
Add check for case when user dosn't have enought collateral to repay the dept.
Add event which is emmited when LiquidationPool is almost or fully depleted.

Add test.
This test shows that in case when user has unsecured debt in excess of his collateral and liquidation pool available liquidity, we don't liquidate him completely, despite the fact that he has collateral.

**Changes in storage or API:**

Describe storage structure changes or API changes. Make sure to update the [API docs](https://minterestfinance.atlassian.net/wiki/spaces/MINTEREST/pages/134807557/Minterest+API).

**Checklist**
- [x] Code is covered with tests
- [ ] Code is covered with benchmarks
- [ ] Fresh benchmark results included

### CODE REVIEW

**Comment by lut1k** [comment]
Could you please remove the extra line of comments: pallets/risk/manager/src/lib.rs line 551

**Comment by lut1k** [comment]
I think that in this PR we should handle the situation on line 430:
https://github.com/minterest-finance/minterest-chain-node/blob/EP/MIN-278-Liquidate-insolvent-loan/pallets/risk-manager/src/lib.rs#L430

File: pallets/risk-manager/src/lib.rs [hunk 1]
```diff
@@ -627,16 +637,22 @@ impl<T: Config> Pallet<T> {
 			}
 		}
 
+		let price_borrowed =
```
Comment (kurotych) [inline pallets/risk-manager/src/lib.rs:640]: Why did you move price_borrowed on top?

File: runtime/src/tests/liquidation_tests.rs [hunk 2]
```diff
@@ -303,17 +283,240 @@ fn liquidation_not_enough_collateral() {
...
+/*
+Description of scenario:
+This scenario handles the case, when user has not enough collateral to cover liquidation and user's
+liquidation attempts are equal to zero.
+This is a rare but possible case (may be caused by Flash Crashes of BTC or outage of oracles).
+The algorithm performs liquidation: liquidation pool can not repay full user's loan, the part of borrow balance and collateral remains with the user,
+DepletedLiquidationPool event is emitted.
+ */
+#[test]
+fn liquidation_by_depleted_pool() {
```
Comment (kurotych) [inline runtime/src/tests/liquidation_tests.rs:303]: It is quite difficult for me to fully understand this test. 
Could you provide more detail comments with mathematical aspects in one place?
Something like here: https://github.com/minterest-finance/minterest-chain-node/blob/master/pallets/mnt-token/src/tests.rs#L467
